lib/maths/CBoostedTreeFactory.cc

```diff
@@ -226,6 +226,7 @@ CBoostedTreeFactory::buildForPredict(core::CDataFrame& frame, std::size_t depend
         }
     });
     this->prepareDataFrameForTrain(frame);
+    m_TreeImpl->computeClassificationWeights(frame);
```

Comment:
Good catch, but I don't think just this can be done. It relies on the data frame having predicted values written into it. I think you'd need to change this so it actually calls `CBoostedTreeImpl::predict` first, wrap the two operations in a `skipIfAfter(CBoostedTreeImpl::E_NotInitialized` and move them after the call to `determineFeatureDataTypes`.

That said do we want to recompute the weights in this case? It seems to me one wouldn't want them to change on calls to predict.

Reply:
We didn't initialize weights before (e.g., we don't restore them from the model definition). I think we are not wrong by doing it since data summarization preserves the distribution of target classes.

Reply:
I guess it doesn't matter since we don't really intend this to be used for prediction except in test, but it seems conceptually dubious to me. As this is just for test I'm happy to leave as is for the time being. As I understand it, this would only be a change on the python side anyway, since we could always extract and supply the original weights. We can always revisit if this proves to be a problem.